src/citrine/ara/variables.py

```diff
@@ -101,3 +110,232 @@ def __init__(self, *,
         self.name = name
         self.headers = headers
         self.template = template
+
+
+class AttributeByTemplateAfterProcessTemplate(Serializable['AttributeByTemplateAfterProcessTemplate'], Variable):
+    """Attribute of an object that is marked by an attribute template and derives from a process
+    marked by a locally unique object template.
+
+    TODO: define "locally-unique"
+
+    Parameters
+    ---------
+    name: str
+        a short human-readable name to use when referencing the variable
+    headers: list[str]
+        sequence of column headers
+    attributeTemplate: LinkByUID
+        attribute template that identifies the attribute to assign to the variable
+    processTemplate: LinkByUid
+        process template that identifies the originating process
+    """
+
+    name = properties.String('name')
+    headers = properties.List(properties.String, 'headers')
+    attributeTemplate = properties.Object(LinkByUID, 'attributeTemplate')
+    processTemplate = properties.Object(LinkByUID, 'processTemplate')
+    typ = properties.String('type', default="attribute_after_process", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["name", "headers", "attributeTemplate", "processTemplate", "typ"]
+
+    def __init__(self, *,
```

Comment:
Why are we leaving room for additional arguments? Do we expect the interface to change? @maxhutch

Reply:
`*` in the middle of the arguments list isn't for position arguments; it is for [keyword-only arguments](https://www.python.org/dev/peps/pep-3102/).

Reply:
That's confusing, since **kwargs uses two asterisks. But regardless, do we expect to add more fields?

Reply:
Not specifically, but possibly.  I'm doing keyword-only arguments everywhere in Ara...